src/components/Search.tsx

```diff
@@ -0,0 +1,21 @@
+import { useState } from "react";
+
+interface Props {
+  intitialSearchText: string;
+  onSearch: (searchText: string) => void;
+}
+export default function Search({ intitialSearchText, onSearch }: Props) {
+  const [searchText, setSearchText] = useState(intitialSearchText);
+
+  return (
+    <div>
+      <input
+        type="text"
+        value={searchText}
+        onChange={(e) => setSearchText(e.target.value)}
+        onKeyDown={(e) => e.key === "Enter" && onSearch(searchText)}
```

Comment:
The onSearch received with props needs to be triggered for borh cases: button click and enter click on input.

Reply:
Changed to form element